### ISSUE TITLE
(🐞) Subscripted `type` with new union syntax in `TypeAlias` incorrectly reports 'Value of type "Type[type]" is not indexable'

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#12978]
```py
from typing import TypeAlias, Union

A: TypeAlias = type[object] | object  # error: Value of type "Type[type]" is not indexable
B = type[object] | object  # error: Value of type "Type[type]" is not indexable
```

[playground](https://mypy-play.net/?mypy=latest&python=3.10&gist=e3676e4db4dbfd4ad61e548392badd7c)

**Comment by AlexWaygood** [comment]
Duplicate of #12392
